### PR TITLE
Leaderboard | Logic

### PR DESCRIPTION
**Pull request by uzairali19**
This branch contains all the logical aspects of the leaderboard. Mostly the user should be able to GET and POST data in the respective API using async and await functions.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -66,15 +51,16 @@ <h1 class="title head-title">Leader Board</h1>
...
-                      type="text"
+                      type="number"
                       placeholder="Your Score"
                     />
                   </div>
                 </div>
                 <div class="field is-grouped is-align-content-center">
                   <div class="control">
-                    <button class="button is-link" type="button">Submit</button>
+                    <button class="button is-link" type="button" id="submit">Submit</button>
```
Comment (HillaryOkello) [inline src/index.html:63]: - [x] The submit button should hit the `POST API` endpoint and post user entries, your submit button does not post the user entry, please check the `postData` function and make sure it works.

File: src/index.html [hunk 2]
```diff
@@ -22,9 +22,7 @@ <h1 class="title head-title">Leader Board</h1>
                   <div class="columns score">
                     <div class="column"><p class="title">Recent Scores</p></div>
                     <div class="column">
-                      <button class="button is-success" type="button">
-                        Refresh
-                      </button>
+                      <button class="button is-success" type="button" id="refresh">Refresh</button>
```
Comment (HillaryOkello) [inline src/index.html:25]: - [x] The refresh button should hit the `GET API` endpoint causing the scores list to change but it seems not to be working. Please check the API functions and make sure they work properly.

**Review by HillaryOkello — CHANGES_REQUESTED** [review]
# Hello @uzairali19, 

Good job so far! You've passed almost all the checklists for this project.
There are a few issues that you still need to work on to go to the next project but you are almost there! You can do it 💪🏻 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Uzair, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**